git_parser.py

```diff
@@ -2,45 +2,94 @@
 git a commit from the github or from the company server and set the commit in the
 Excel file
 '''
-from datetime import datetime
-
+from datetime import date, timedelta,datetime
 from pydriller import Repository
 from pydriller.domain.commit import ModificationType
 import xlsxwriter as xw
-#import pandas as pd
+import pandas as pd
+
+
+
+def show_commit(args):
 
+    #Create a output file
+    if args.output_file:
+        #Choose a new local path
+        Outputdata = args.output_file +"\outputExcel_file.xlsx"
+        wb = xw.Workbook(Outputdata)
+        print("\nExcel file created in: " + args.output_file + "\outputExcel_file.xlsx")
+    else:
+        #Set in a same arg_parser.py path
+        Outputdata = "outputExcel_file.xlsx"
+        wb = xw.Workbook(Outputdata)
+        print("\nExcel file created in: local_python_file_path\outputExcel_file.xlsx")
 
-wb = xw.Workbook("Excel_File1.xlsx")
-ws = wb.add_worksheet(name="Git_commit")
+    #Set worksheet name
+    ws = wb.add_worksheet(name="parser_git_commit")
 
-firstDay = datetime(2022, 5, 1)
-toDay = datetime.now()
+    #Add worksheet font format
+    bold = wb.add_format({'bold': True})
+    italic = wb.add_format(dict(italic=True))
 
+    #Calculate date
+    to_day = datetime.now().today()
 
-bold = wb.add_format({'bold': True})
-italic = wb.add_format(dict(italic=True))
+    if args.show_lats_days:
 
+        first_day = to_day - timedelta(days=args.show_lats_days)
+        print(first_day)
+        print(to_day)
 
-def show_commit(url):
+    else:
+        first_day = datetime.min
+        print(to_day)
+
+
+    print("Please wait...")
     Row_number = 0
     Col_number = 0
 
-    ws.write(Row_number, Col_number, 'File Name', bold)
-    ws.write(Row_number, Col_number + 1, 'Commit msg', bold)
+    ws.write(Row_number, Col_number, 'hash commit', bold)
+    ws.write(Row_number, Col_number + 1, 'Commit subject', bold)
     ws.write(Row_number, Col_number + 2, 'Commit Date', bold)
-    ws.write(Row_number, Col_number + 3, 'hash commit', bold)
+    ws.write(Row_number, Col_number + 3, 'File Name', bold)
 
-    Row_number += 1
+    hash_commit = 'xxxx'
 
-    for commit in Repository(path_to_repo=url, since=firstDay, to=toDay).traverse_commits():
+    for commit in Repository(path_to_repo=args.repository, since=first_day, to=to_day).traverse_commits():
         for modified_file in commit.modified_files:
```

Comment:
remove this  for if else, modified_files is a list of strings, put all file names in the same column.